htdocs/compta/facture/class/facture.class.php

```diff
@@ -1520,8 +1520,12 @@ public function getNomUrl($withpicto = 0, $option = '', $max = 0, $short = 0, $m
 			if ($this->type == self::TYPE_SITUATION) {
 				$label = img_picto('', $picto).' <u class="paddingrightonly">'.$langs->transnoentitiesnoconv("InvoiceSituation").'</u>';
 			}
-			if (isset($this->statut) && isset($this->alreadypaid)) {
-				$label .= ' '.$this->getLibStatut(5, $this->alreadypaid);
+			if (isset($this->statut)) {
+				if (!isset($this->alreadypaid)) {
+					$this->alreadypaid = $this->getSommePaiement();
```

Comment:
Having a call of getSommePaiement that make heavy database access inside a method dedicated to output formatting is  a problem. The pro is that it solves trouble everywhere. The cons is that this need a high cost in performance, and does not match good practices: no database access for formatting output.
Can you try to make the getSommePaiement on the caller side (when on caller side, we often have already the data or we will be able later to use a cache for each invoice, so cost will be paid only if required).